api/include/opentelemetry/context/context.h

```diff
@@ -0,0 +1,205 @@
+#pragma once
+
+#include <atomic>
+
+#include "opentelemetry/context/context_value.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/key_value_iterable_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The context class provides a context identifier. Is built as a linked list
+// of context nodes which each have access to the nodes before it, but not 
+// after. So each time a write is performed a new context is added to the list.
+// Within each context node there is another linked list which holds the key
+// and value data for each node.
+class Context
+{
+
+public:
+  
+  // A linked list to contain the keys and values of this context node 
+  class DataList {
+    private:
+      friend class Context;
+ 
+      DataList *next_;
+      
+      char key_[50];
```

Comment:
Have you thought about how to handle cases where the key is larger than 50 characters?

I'm wondering if it might be easier to just allocate space for the data dynamically on the heap.

Reply:
changed to dynamic char *.